### ISSUE TITLE
Store a sha256sum in S3 object metadata; use it for comparison later

### ISSUE THREAD

**Issue by tsibley** [nextstrain/ncov-ingest#35]
…instead of downloading the whole object every time to hash it inside `upload-to-s3`.

Storing the sha256sum can use the `--metadata sha256sum=...` flag of `aws s3 cp`.

Retrieving the sha256sum can use something like `aws s3api head-object … | jq -r .Metadata.sha256sum`.

**Comment by kairstenfay** [comment]
One thought here: 

`aws s3api head-object` requires both a `--bucket` (no s3:// prefix) and a `--key` (file name). Would the best way to get these be to split the string provided for the `dst` argument?

**Comment by tsibley** [comment]
Yep, split would work, or bashisms like:

```
s3path="${url#s3://}"
bucket="${s3path%%/*}"
key="${s3path#*/}"
```

**Comment by kairstenfay** [comment]
> Yep, split would work, or bashisms like:
> 
> ```
> s3path="${url#s3://}"
> bucket="${s3path%%/*}"
> key="${s3path#*/}"
> ```

@tsibley What is this called in Bash? Pattern matching? I'm wondering how to find the documentation for these special characters like `#`, `%`, etc.

**Comment by kairstenfay** [comment]
Closed via #37

**Comment by tsibley** [comment]
This is part of the _parameter expansion_ syntax `${…}`. See that subsection under _Expansion_ in the [bash man page](http://man7.org/linux/man-pages/man1/bash.1.html#EXPANSION) or the equivalent section of the [full manual](https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion).
